crates/store/src/db/sql.rs

```diff
@@ -718,6 +770,31 @@ pub fn select_block_header_by_block_num(
     }
 }
 
+/// Select all the given block headers from the DB using the given [Connection].
+///
+/// # Returns
+///
+/// A vector of [BlockHeader] or an error.
+pub fn select_many_block_headers(
+    conn: &mut Connection,
+    blocks: Vec<BlockNumber>,
+) -> Result<Vec<BlockHeader>> {
+    let mut stmt =
+        conn.prepare_cached("SELECT block_header FROM block_headers WHERE block_num = ?1;")?;
+    let mut result = Vec::with_capacity(blocks.len());
+
+    for block in blocks {
+        let header: Vec<u8> = stmt
+            .query_row([block], |row| row.get(0))
+            .optional()?
+            .ok_or(DatabaseError::BlockNotFoundInDb(block))?;
+        let header = BlockHeader::read_from_bytes(&header)?;
+        result.push(header);
+    }
```

Comment:
You're right, but I had concerns with the output not aligning with the input since the input is a slice and not a set.

And selecting using `IN` would return a set.

I'll see if I can't improve this and the function parameters.

Reply:
You mean that `blocks` parameter could contain duplicate block numbers? I think we should think of this as an error - though not sure we should check for it (at least at this level).

I'm also not sure we should enforce that for every passed-in block number there must be a record in the DB. So, I think this is OK that this function returns a vector of different size as compared to the passed-in vector?

Basically, the way I think of this function is that given a list of block numbers, it returns a set of matching block headers from the database. But it can also return no results if none block headers match.

Reply:
And to add: it would then be a responsibility of the caller to make sense of these results.